COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs

```diff
@@ -0,0 +1,144 @@
+        /// initial parameters from its parent in the render tree.
+        /// </summary>
+        protected override void OnInitialized()
+        {
+            base.OnInitialized();
+
+            if (this.ParameterType is ArrayParameterType arrayParameterType)
+            {
+                this.Dimension = string.Join(",", arrayParameterType.Dimension.Select(x => x.ToString()));
+            }
+        }
+
+        /// <summary>
+        /// Gets the available scales based on the <see cref="ParameterType" /> from <see cref="ParameterTypeComponent" />
+        /// </summary>
+        /// <returns>A collection of the available scales</returns>
+        private IEnumerable<MeasurementScale> GetAvailableScales()
+        {
+            return this.Item.ParameterType is not QuantityKind quantityKind ? Enumerable.Empty<MeasurementScale>() : quantityKind.AllPossibleScale.OrderBy(x => x.Name);
+        }
+
+        /// <summary>
+        /// Method invoked when creating a new parameter type component
+        /// </summary>
+        /// <param name="e">A <see cref="GridCustomizeEditModelEventArgs" /></param>
+        private void CustomizeEditParameterTypeComponent(GridCustomizeEditModelEventArgs e)
+        {
+            var dataItem = (ParameterTypeComponentRowViewModel)e.DataItem;
+            this.ShouldCreate = e.IsNew;
+
+            this.Item = dataItem == null
+                ? new ParameterTypeComponent { Iid = Guid.NewGuid() }
+                : dataItem.Thing.Clone(true);
+
+            e.EditModel = this.Item;
+        }
+
+        /// <summary>
+        /// Method executed every time the dimension field text has changed
+        /// </summary>
+        /// <param name="text">The new text</param>
+        private void OnTextChanged(string text)
+        {
+            this.Dimension = text;
+            var dimensions = text.Split(",").Select(int.Parse).ToList();
+
+            var arrayParameterType = (ArrayParameterType)this.ParameterType;
+
+            if (arrayParameterType.Dimension.Count > dimensions.Count)
+            {
+                var j = 0;
+
+                foreach (var dimension in arrayParameterType.Dimension.ToList())
+                {
+                    if (j >= dimensions.Count)
+                    {
+                        arrayParameterType.Dimension.Remove(dimension);
+                    }
+
+                    j++;
+                }
+            }
+
+            if (arrayParameterType.Dimension.Count >= dimensions.Count)
+            {
+                return;
+            }
+
+            var k = 0;
+
+            foreach (var dimension in dimensions)
+            {
+                if (k >= arrayParameterType.Dimension.Count)
+                {
+                    arrayParameterType.Dimension.Add(dimension);
+                }
+
+                k++;
+            }
+
+            //update the contained component according with the updated dimensions
```

Comment:
I assume that WIP here?

Reply:
Yeah, Sam is going to create a new ticket for that. It's about the implementation of ArrayParameterType.

Reply:
see https://github.com/STARIONGROUP/COMET-WEB-Community-Edition/issues/602